frb_codegen/src/generator/dart/ty_record.rs

```diff
@@ -0,0 +1,72 @@
+use crate::generator::dart::ty::*;
+use crate::ir::*;
+use crate::target::Acc;
+use crate::type_dart_generator_struct;
+
+type_dart_generator_struct!(TypeRecordGenerator, IrTypeRecord);
+
+impl TypeDartGeneratorTrait for TypeRecordGenerator<'_> {
+    fn api2wire_body(&self) -> Acc<Option<String>> {
+        Acc {
+            wasm: self.context.config.wasm_enabled.then(|| {
+                let values = self
+                    .ir
+                    .values
+                    .iter()
+                    .enumerate()
+                    .map(|(idx, value)| {
+                        format!("api2wire_{}(raw.${})", value.safe_ident(), idx + 1)
+                    })
+                    .collect::<Vec<_>>()
+                    .join(",");
+                format!("return [{values}];")
+            }),
+            ..Default::default()
+        }
+    }
+
+    fn api_fill_to_wire_body(&self) -> Option<String> {
+        let values = self
+            .ir
+            .values
+            .iter()
+            .enumerate()
+            .map(|(idx, value)| {
+                if value.is_struct() {
+                    format!(
+                        "_api_fill_to_wire_{}(apiObj.${}, wireObj.field{idx});",
+                        value.safe_ident(),
+                        idx + 1
+                    )
+                } else {
+                    format!(
+                        "wireObj.field{idx} = api2wire_{}(apiObj.${});",
+                        value.safe_ident(),
+                        idx + 1
+                    )
+                }
```

Comment:
hmm is it possible we unify the two cases? 

Or, does this pattern occur repeatedly? If so can we extract some common code?

Reply:
This was copied verbatim from IrStruct, so perhaps a solution for it may be desired as well.

Reply:
👀 Seems we forget to update this? It is still duplicated there

Reply:
Record fields are accessed using the `${index + 1}` syntax compared to our homegrown `field$index`, so they aren't compatible and had to be brought over from ty_struct.

Reply:
Looking at https://github.com/fzyzcjy/flutter_rust_bridge/blob/b7ec2dabe9f6582c3d544f0e1239d3cba18b3dfb/frb_codegen/src/generator/dart/ty_struct.rs#L42-L56, what about this:

```rust
fn f(api_obj_field_name: &str) {
                    if field.ty.is_struct() {
                        format!(
                            "_api_fill_to_wire_{}(apiObj.{}, wireObj.{});",
                            field.ty.safe_ident(),
                            api_obj_field_name,
                            field.name.rust_style(),
                        )
                    } else {
                        format!(
                            "wireObj.{} = api2wire_{}(apiObj.{});",
                            field.name.rust_style(),
                            field.ty.safe_ident(),
                            api_obj_field_name,
                        )
                    }
}
```

where api_obj_field_name is `${idx+1}` here and `field.name.dart_style()` there

Reply:
I could merge that code, but personally I think that introduces a bit of unneeded coupling between the two modules and might end up being split again in the future, but up to you.

Reply:
Hmm I am not sure. We already introduced coupling between Record and Struct in rust/ty_record.rs. So for consistency, I guess it is either acceptable to couple in dart/ty_record.rs, or we should also avoid that in the rust/ty_record.rs?